src/components/Divider.tsx

```diff
@@ -1,70 +1,141 @@
+        })
+      }
+
+      return {
+        isOnlyChild,
+        isTableView,
+        children,
+        thoughtsAtSameDepth,
+      }
+    },
+    // Ensure the selector runs only when values change, excluding ref changes to avoid redundant re-renders.
+    (prev, next) =>
+      prev.isOnlyChild === next.isOnlyChild &&
+      prev.isTableView === next.isTableView &&
+      prev.children.length === next.children.length &&
+      prev.children.every((child, index) => child.value === next.children[index].value) &&
+      prev.thoughtsAtSameDepth.length === next.thoughtsAtSameDepth.length &&
+      prev.thoughtsAtSameDepth.every((thought, index) => thought.value === next.thoughtsAtSameDepth[index].value),
+  )
+}
+
+/**
+ * Calculates the width of multiple thoughts by measuring their rendered widths in the DOM.
+ */
+const getThoughtWidths = (thoughts: { id: ThoughtId }[]): number[] => {
+  return thoughts.map(thought => {
+    const innerThoughtElement = document.querySelector(
+      `[aria-label="tree-node"][data-id="${thought.id}"] [aria-label="thought"]`,
+    )
+    return innerThoughtElement?.getBoundingClientRect().width ?? 0
+  })
+}
 
 /** A custom horizontal rule. */
 const Divider = ({ path, cssRaw }: { path: Path; cssRaw?: SystemStyleObject }) => {
-  const dividerSetWidth = React.createRef<HTMLInputElement>()
   const dispatch = useDispatch()
+  const dividerRef = useRef<HTMLDivElement>(null)
+  const [dividerWidth, setDividerWidth] = useState<number>(DIVIDER_MIN_WIDTH)
+  const { isOnlyChild, isTableView, children, thoughtsAtSameDepth } = useDividerData(path)
+  const editingThoughtId = useSelector((state: State) => state.cursor && head(state.cursor))
+  const editingValueUntrimmed = editingValueUntrimmedStore.useSelector(editingValue => editingValue)
 
   /** Sets the cursor to the divider. */
   const setCursorToDivider = (e: React.MouseEvent | React.TouchEvent) => {
     e.stopPropagation()
     dispatch(setCursor({ path }))
   }
 
-  /** Get the max width of nearby for divider list child elements, add 30 px and set this width for divider. */
-  const setStyle = () => {
-    if (dividerSetWidth.current) {
-      const parentUl = dividerSetWidth.current.closest('ul')
-      const children = parentUl ? (Array.from(parentUl.childNodes) as HTMLElement[]) : []
-      const widths = children.map((child: HTMLElement) => {
-        if (child.querySelector('[data-divider=true]')) return DIVIDER_PLUS_PX
-        const subs = child.getElementsByClassName('subthought') as HTMLCollectionOf<HTMLElement>
-        return subs.length ? subs[0].offsetWidth + DIVIDER_PLUS_PX : DIVIDER_PLUS_PX
-      })
-      const maxWidth = Math.max(...widths)
-      dividerSetWidth.current.style.width = `${maxWidth > DIVIDER_MIN_WIDTH ? maxWidth : DIVIDER_MIN_WIDTH}px`
-    }
-  }
+  useEffect(() => {
+    /** Calculates and updates the Divider's width based on sibling thought widths. */
+    const updateDividerWidth = () => {
```

Comment:
The multiple return statements and calls to `setDivideWidth` make this function look more complicated than it actually is. Try refactoring it so that there is a single `setDividerWidth` at the end and no early returns. `Math.max(...widths, DIVIDER_MIN_WIDTH))` naturally returns `DIVIDER_MIN_WIDTH` when `widths = []`.